### PR TITLE
Change date version + add release information

### PR DESCRIPTION
**Pull request by dhoko**
```json
{
  "version": "3.16.20",
  "commit": "08f231e7c5e5eea9f08b8c165ef13fb317836627",
  "branch": "heads/v4",
  "buildDate": "2020-01-21T15:11:08CET",
  "release": "heads/v4-0-g08f231e7c"
}
```

### CODE REVIEW

File: scripts/createVersionJSON.sh [hunk 1]
```diff
@@ -52,12 +52,13 @@ function getBranch {
 }
 
 function toJSON {
-    local tpl='{ "version": "#version", "commit": "#commit", "branch": "#branch", "date": "#date" }';
+    local tpl='{ "version": "#version", "commit": "#commit", "branch": "#branch", "buildDate": "#date", "release": "#release" }';
     local commit=$(getCommit);
     local version=$(getVersion);
     local branch=$(getBranch);
```
Comment (jpbochi) [inline scripts/createVersionJSON.sh:57]: branch was already there? o_O

File: scripts/createVersionJSON.sh [hunk 2]
```diff
@@ -52,12 +52,13 @@ function getBranch {
...
-    local date="$(date '+%s')";
-    echo "$tpl" | sed "s/#commit/$commit/; s/#version/$version/; s|#branch|$branch|; s|#date|$date|;"
+    local date="$(date '+%FT%T%Z')";
+    local release="$(git describe --long --dirty --all)";
+    echo "$tpl" | sed "s/#commit/$commit/; s/#version/$version/; s|#branch|$branch|; s|#date|$date|; s|#release|$release|;"
```
Comment (jpbochi) [inline scripts/createVersionJSON.sh]: some of these use `/` and others use `|`. I think I know why. One safer (yet a bit more complicated) alternative is `envsubst`. Here are a couple of examples:

https://github.com/ProtonMail/Angular/blob/v4/docker/entrypoint.sh#L9
https://gitlab.protontech.ch/proton/devops/test-infra/blob/master/kube-get-current-versions.sh#L26

File: scripts/createVersionJSON.sh [hunk 3]
```diff
@@ -52,12 +52,13 @@ function getBranch {
...
 
 function toJSON {
-    local tpl='{ "version": "#version", "commit": "#commit", "branch": "#branch", "date": "#date" }';
+    local tpl='{ "version": "#version", "commit": "#commit", "branch": "#branch", "buildDate": "#date", "release": "#release" }';
     local commit=$(getCommit);
     local version=$(getVersion);
     local branch=$(getBranch);
-    local date="$(date '+%s')";
-    echo "$tpl" | sed "s/#commit/$commit/; s/#version/$version/; s|#branch|$branch|; s|#date|$date|;"
+    local date="$(date '+%FT%T%Z')";
```
Comment (jpbochi) [inline scripts/createVersionJSON.sh]: I think we should go with UTC, and do `date -u '+%FT%TZ'`.

File: scripts/createVersionJSON.sh (hunk 2 above)
Comment (dhoko) [reply scripts/createVersionJSON.sh]: I  wasn't able to make it run :/ so instead I use cat :grin:

Comment (jpbochi) [reply scripts/createVersionJSON.sh]: 🆒

**Pull request merged**